test/data_tests/cassiopeia_tree_test.py

```diff
@@ -1664,6 +1664,40 @@ def test_set_dissimilarity(self):
                 "node3", {leaf: 0 for leaf in tree_no_map.leaves}
             )
 
+    def test_scale_to_unit_length(self):
```

Comment:
Can you add an additional test to verify that `get_max_depth_of_tree` works as described now?

Reply:
Great point. Now `get_max_depth_of_tree` and `get_mean_depth_of_tree` should be correct and tested for our border case. I modified the already existing `test_depth_calculations_on_tree` test to accomplish this.